ArmEmulator/Mem.fs

```diff
@@ -107,79 +122,62 @@ module Mem
             match ops with
             | Match @"(R[0-9]|1[0-5]) *, *\[ *(R[0-9]|1[0-5]) *] *, *#([0-9]+)" [_; rA; rB; incV] -> //Post Increment
-                Ok ({Ra=regNamesTryFindMonad rA; Rb=regNamesTryFindMonad rB; IncrVal=(incV.Value|>int); Post= true; Rc= None; Shift= None;})
+                Ok {(defaultRecord rA rB) with IncrVal=(incV.Value|>int); Post= true;}
             | Match @"(R[0-9]|R1[0-5]) *, *\[ *(R[0-9]|R1[0-5]) *]" [_; rA; rB] -> //Base Case
-                Ok ({Ra=regNamesTryFindMonad rA; Rb=regNamesTryFindMonad rB; IncrVal=0; Post= false; Rc= None; Shift= None;})
+                Ok (defaultRecord rA rB)
             | Match @"(R[0-9]|R1[0-5]) *, *\[ *(R[0-9]|R1[0-5]) *, *#([0-9]+) *\]" [_; rA; rB; incV] -> //Num Increment
-                Ok ({Ra=regNamesTryFindMonad rA; Rb=regNamesTryFindMonad rB; IncrVal=(incV.Value|>int); Post= false; Rc= None; Shift= None;})
+                Ok {(defaultRecord rA rB) with IncrVal=(incV.Value|>int);}
             | Match @"(R[0-9]|R1[0-5]) *, *\[ *(R[0-9]|R1[0-5]) *, *(R[0-9]|R1[0-5]) *\]" [_; rA; rB; rC] -> //Adding Registers
-                Ok ({Ra=regNamesTryFindMonad rA; Rb=regNamesTryFindMonad rB; IncrVal=0; Post= false; Rc= Some regNames.[rC.Value]; Shift= None;})
+                Ok {(defaultRecord rA rB) with Rc= Some regNames.[rC.Value];}
             | Match @"(R[0-9]|R1[0-5]) *, *\[ *(R[0-9]|R1[0-5]) *, *(R[0-9]|R1[0-5]) *, *LSL *#(-*[0-9]+) *\]" [_; rA; rB; rC0; shft0] -> //Shifting
                 let shft, rC = isPos shft0 rC0
-                Ok ({Ra=regNamesTryFindMonad rA; Rb=regNamesTryFindMonad rB; IncrVal=0; Post= false; Rc= rC; Shift= shft;})
+                Ok {(defaultRecord rA rB) with Rc= rC; Shift= shft;}
             | _ -> 
                 Error (sprintf "ops didn't match anything\nops: %A\n" ops)
 
-        let makeOutFromParseOps (x: FromOps) _ =
+        let makeOutFromParseOps _ (x: FromOps) =
             {InstructionType= instTypeTmp;
                 DestSourceReg= x.Ra; AddressReg= x.Rb;
                 BytesNotWords= bytes; IncrementValue= x.IncrVal;
                 PreIndexRb= pre; PostIndexRb= x.Post; 
                 ExtraAddressReg= x.Rc;
                 ShiftExtraRegBy= x.Shift;}
-        resultDotBindTwoInp makeOutFromParseOps (parseOps ((ls.Operands).Trim())) errorMessage1 
-
+        parseOps ((ls.Operands).Trim())
+        |> resultDotBindTwoInp makeOutFromParseOps errorMessage1
         
 
 
 
 
-
-
-
-
-
-
-
-
 //----------ADR INSTRUCTION DEFINITION AND PARSING-------------------------------------------------
 
-    type ADRInstrType = ADRm 
+    type ADRInstrType = ADRm
+    //The m is needed to it doesn't interfere with ADR of
+    // type InstrClass in Common Lex
 
 
-    /// instruction (dummy: must change)
+    ///A record which encapsulates all the necessary 
+    /// information for executing an ADR instruction
     type ADRInstr =
         {
             InstructionType: Result<ADRInstrType,String>;
```

Comment:
It's fine for now but I think these shouldn't be Result monads. You should just bind/map the result at every opportunity so it gets drawn all the way out to the top.